external-crates/move/crates/move-vm-types/src/values/values_impl.rs

```diff
@@ -4202,4 +4204,138 @@ impl Value {
     pub fn as_move_value(&self, layout: &MoveTypeLayout) -> MoveValue {
         self.0.as_move_value(layout)
     }
+
+    #[cfg(feature = "gas-profiler")]
+    pub fn as_annotated_move_value(&self, layout: &A::MoveTypeLayout) -> Option<A::MoveValue> {
+        self.0.as_annotated_move_value(layout)
+    }
+}
+
+#[cfg(feature = "gas-profiler")]
```

Comment:
What is the motivation for feature gating these?

Reply:
A couple reasons:
1. They are only used for move tracing, and in general we should be careful about when and how we create annotated values since they can inflate the value size.
2. This is snapshotting the value to an annotated value -- in particular it goes through references. Whereas in a "true" VM value -> annotated value function you may want this to be an error.